File: 739.每日温度/solution.cpp
# include "..\leetcode.h"

class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        // 单调栈，存储下标
        stack<int> st;
        vector<int> ans(temperatures.size(), 0);
        for(int i = 0; i < temperatures.size(); i++) {
            while(!st.empty() && temperatures[i] > temperatures[st.top()]) {
                ans[st.top()] = i - st.top();
                st.pop();
            }
            st.push(i);
        }
        return ans;
    }
};

File: 739.每日温度/solution.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        m = len(temperatures)
        st = []
        ans = [0] * m
        for i in range(m):
            while len(st) != 0 and temperatures[i] > temperatures[st[-1]]:
                ans[st[-1]] = i - st[-1]
                st.pop()
            st.append(i)
        return ans
